src/engine/threaded_engine.h

```diff
@@ -338,33 +346,46 @@ class ThreadedEngine : public Engine {
 #endif
     CallbackOnComplete callback = this->CreateCallback(
         ThreadedEngine::OnCompleteStatic, opr_block);
+    CallbackOnComplete on_start_callback = this->CreateCallback(
```

Comment:
what's the point of creating a call back here?

Reply:
The callback is not strictly necessary here, since it is called only once, but I included it to keep the ExecuteOprBlock easier to read and separate out the OnStart logic.

Reply:
I think this is unnecessary overhead. Call OnStart directly if possible

Reply:
Called OnStart directly.